lib/helpers/calculator_helpers.js

```diff
@@ -0,0 +1,469 @@
+            // Add the single result value to the values array on the results of
+            // this calculation (allowing for more than one possible observation).
+            if (obsResult != null ? Object.prototype.hasOwnProperty.call(obsResult, 'value') : undefined) {
+              // If result is a Cql.Quantity type, add its value
+              resultValue = obsResult.observation.value;
+            } else {
+              // In all other cases, add result
+              resultValue = obsResult.observation;
+            }
+
+            // if the episodeResult object already exist create or add to to the values structure
+            if (episodeResults[episodeId] != null) {
+              if (episodeResults[episodeId].values != null) {
+                episodeResults[episodeId].values.push(resultValue);
+              } else {
+                episodeResults[episodeId].values = [resultValue];
+              }
+              // else create a new episodeResult structure
+            } else {
+              newEpisode = {};
+              popCodesInPopulation.forEach((pop) => {
+                if (pop !== 'OBSERV') {
+                  newEpisode[pop] = 0;
+                }
+              });
+              newEpisode.values = [resultValue];
+              episodeResults[episodeId] = newEpisode;
+            }
+          });
+        });
+      }
+    });
+
+    // Correct any inconsistencies. ex. In DENEX but also in NUMER using same function used for patients.
+    Object.keys(episodeResults).forEach((episodeId) => {
+      const episodeResult = episodeResults[episodeId];
+      episodeResults[episodeId] = this.handlePopulationValues(episodeResult);
+    });
+
+    return episodeResults;
+  }
+
+  // Set all value set versions to 'undefined' so the execution engine does not grab the specified version in the ELM
+  static setValueSetVersionsToUndefined(elm) {
+    Array.from(elm).forEach((elmLibrary) => {
+      if (elmLibrary.library.valueSets != null) {
+        Array.from(elmLibrary.library.valueSets.def).forEach((valueSet) => {
+          if (valueSet.version != null) {
+            valueSet.version = undefined;
+          }
+        });
+      }
+    });
+    return elm;
+  }
+
+  // Format ValueSets for use by the execution engine
+  static valueSetsForCodeService(valueSetsByOid, valueSetOids = []) {
+    const valueSets = {};
+    valueSetOids.forEach((oid) => {
+      let specificValueSet = valueSetsByOid[oid];
+      if (!(specificValueSet && !specificValueSet[Object.keys(specificValueSet)[0]].concepts)) {
+        specificValueSet = specificValueSet[Object.keys(specificValueSet)[0]];
+      }
+      if (!valueSets[specificValueSet.oid]) {
+        valueSets[specificValueSet.oid] = {};
+      }
+      if (!valueSets[specificValueSet.oid][specificValueSet.version]) {
+        valueSets[specificValueSet.oid][specificValueSet.version] = [];
+      }
+      specificValueSet.concepts.forEach((concept) => {
```

Comment:
A check needs to be added back into this section to confirm that `concepts` exist. 

This code existed on the frontend version:  `continue unless specific_value_set.concepts`